### DISCUSSION TITLE
Is there a way to avoid the edge created when restoring face?

### DISCUSSION THREAD

**Discussion by bconnor218** [glucauze/sd-webui-faceswaplab#122] [Q&A]
I've noticed when performing a swap that in order for the face to look clear I have to perform a restore face.  however the restore face creates an edge around the face which is more sharp than the rest of the image.  If I do an inpaint post face swap and set denoise to .5 this clears helps it, but it tends to change the face a bit too.

Is there a way that I'm missing, similar to img2img inpaint, that could blur the mask around the face so that there isn't such a defined edge?

**Comment by glucauze** [comment]
Are you refering to that : https://github.com/glucauze/sd-webui-faceswaplab/issues/38 ?

**Reply by bconnor218** [reply]
No.  I'm referring to the box that created when restoring a face by Codeformer or GFPGAN.  It appears to restore a beyond the face and when there are things like hair in the picture it results in some of the hair or other elements being sharper than the rest of the image.  There is a noticeable edge around the face as a result.

**Comment by glucauze** [comment]
In that case i suggest to use improved mask, an upscaler and codeformer in the settings. It should mitigate that problem.

See https://github.com/glucauze/sd-webui-faceswaplab/discussions/50
